diff --git a/src/neo4j.js b/src/neo4j.js
index a0a886c..7fdd84e 100644
--- a/src/neo4j.js
+++ b/src/neo4j.js
@@ -1,4 +1,4 @@
-// TODO: Import the neo4j-driver dependency
+import neo4j from 'neo4j-driver'
 
 /**
  * A singleton instance of the Neo4j Driver to be used across the app
@@ -20,7 +20,17 @@ let driver
  */
 // tag::initDriver[]
 export async function initDriver(uri, username, password) {
-  // TODO: Create an instance of the driver here
+  driver = neo4j.driver(
+    uri,
+    neo4j.auth.basic(
+      username,
+      password
+    )
+  )
+
+  await driver.verifyConnectivity()
+
+  return driver
 }
 // end::initDriver[]
 
