diff --git a/src/services/rating.service.js b/src/services/rating.service.js
index 8c481ab..a81e5d9 100644
--- a/src/services/rating.service.js
+++ b/src/services/rating.service.js
@@ -40,9 +40,30 @@ export default class ReviewService {
    */
   // tag::forMovie[]
   async forMovie(id, sort = 'timestamp', order = 'ASC', limit = 6, skip = 0) {
-    // TODO: Get ratings for a Movie
+    // Open a new database session
+    const session = this.driver.session()
+
+    // Get ratings for a Movie
+    const res = await session.executeRead(
+      tx => tx.run(`
+        MATCH (u:User)-[r:RATED]->(m:Movie {tmdbId: $id})
+        RETURN r {
+          .rating,
+          .timestamp,
+          user: u {
+            .id, .name
+          }
+        } AS review
+        ORDER BY r.\`${sort}\` ${order}
+        SKIP $skip
+        LIMIT $limit
+      `, { id, limit: int(limit), skip: int(skip) })
+    )
+
+    // Close the session
+    await session.close()
 
-    return ratings
+    return res.records.map(row => toNativeTypes(row.get('review')))
   }
   // end::forMovie[]
 
